components/fxa-client/src/migrator.rs

```diff
@@ -24,40 +26,72 @@ impl FirefoxAccount {
         k_sync: &str,
         k_xcs: &str,
         copy_session_token: bool,
-    ) -> Result<()> {
+    ) -> Result<bool> {
         // if there is already a session token on account, we error out.
         if self.state.session_token.is_some() {
             return Err(ErrorKind::IllegalState("Session Token is already set.").into());
         }
 
-        let migration_session_token = if copy_session_token {
-            let duplicate_session = self
-                .client
-                .duplicate_session(&self.state.config, &session_token)?;
+        self.state.in_flight_migration = Some(MigrationData {
+            k_sync: k_sync.to_string(),
+            k_xcs: k_xcs.to_string(),
+            copy_session_token,
+            session_token: session_token.to_string(),
+        });
+
+        let client = self.client.clone();
+        match self.helper_migration_network_methods(client) {
+            Ok(_) => {
+                log::info!("FxA migration complete");
+                Ok(true)
+            }
+            Err(err) => {
+                log::info!("Failed to perform network requests: {}", err);
+                // don't throw hard error here, let the consumers try again
```

Comment:
This comment may be out of date, since it's not the consumer that tries again, we try again ourselves internally.